### PR TITLE
[model] feat: support NPU wan2.2_I2V_5B/14B

### PR DESCRIPTION
**Pull request by dzy00897185**
### What does this PR do?

Adds Ascend NPU support for wan2.2_I2V_5B/14B training. First, training-related YAML and JSON configuration files are added under VeOmni/configs/dit. Based on the original T2V training codebase, modifications such as additional input channels are implemented to accommodate the new I2V logic. 

### Checklist Before Starting

- Search for relative PRs/issues and link here: ...
- PR title follows `[{modules}] {type}: {description}` format (enforced by [check_pr_title.yml](.github/workflows/check_pr_title.yml))
  - **Allowed modules:** `agent`, `ci`, `ckpt`, `config`, `data`, `dist`, `docker`, `docs`, `logging`, `lora`, `misc`, `model`, `omni`, `optim`, `ops`, `parallel`, `perf`, `release`, `task`, `trainer`
  - **Allowed types:** `feat`, `fix`, `refactor`, `chore`, `test`
  - Breaking changes: prepend `[BREAKING]` — e.g. `[BREAKING][parallel, model] feat: dynamic batching`

### Test

> Validation results (training curves, eval metrics) for changes not covered by CI.

### API and Usage Example

> Show API changes and usage examples if applicable.

### Design & Code Changes
.agents/knowledge/architecture.md
configs/dit/wan2.2_I2V_5B.yaml
configs/model_configs/wan/wan2.2_i2v_5b.json
configs/text/qwen3‑moe.yaml
veomni/arguments/arguments_types.py
veomni/distributed/clip_grad_norm.py
veomni/distributed/torch_parallelize.py
veomni/models/diffusers/wan_t2v/wan_condition/configuration_wan_condition.py
veomni/models/diffusers/wan_t2v/wan_transformer/modeling_wan_transformer.py
veomni/trainer/base.py
veomni/trainer/callbacks/base.py
veomni/trainer/callbacks/channel_loss_callback.py
veomni/trainer/callbacks/trace_callback.py
veomni/utils/device.py
configs/dit/wan2.2_I2V_14B.yaml
configs/model_configs/wan/wan2.2_i2v_14b.json
docs/key_features/async_activate_offload.md
tests/distributed/test_async_activation_offload.py
veomni/distributed/async_offloading.py

> High-level design description and specific change list.

### Checklist Before Submitting

- Read the [Contribute Guide](https://github.com/ByteDance-Seed/VeOmni/blob/main/CONTRIBUTING.md)
- Applied pre-commit checks
- Added/updated documentation
- If `tasks/` training scripts were moved or renamed: updated `docs/` examples and verified `python3 scripts/ci/check_doc_task_paths.py` passes (also enforced by the **Check doc task paths** CI workflow)
- Added tests to CI workflow (or explained why not feasible)

## Summary by CodeRabbit

- **New Features**
  - Added Wan 2.2 image-to-video training support for 5B and 14B models.
  - Added asynchronous activation offloading with prefetching to reduce accelerator memory usage.
  - Added expanded timestep conditioning and improved image-latent handling.
  - Added broader NPU and device compatibility.

- **Bug Fixes**
  - Improved gradient norm handling across training callbacks.
  - Improved configuration loading, validation, and model input handling.

- **Documentation**
  - Added comprehensive guidance for asynchronous activation offloading.

### CODE REVIEW

**Comment by Coach257** [comment]
Please use async offload in #965

**Comment by dzy00897185** [comment]
> 请在#965中使用异步卸载

之前的异步offload并不适配

**Review by Coach257 — CHANGES_REQUESTED** [review]
@dzy00897185 On “please use the async offload in #965; the previous one does not fit” — agreed, but this branch is still not wired to #965. After merging `main`, the two implementations are stacked and the training path is broken. Please follow the steps below and stop maintaining a separate offload stack.

## What is wrong today

`veomni/distributed/async_offload.py` (#965) was not rewritten; it is identical to `main`. This PR instead:

1. Added a second implementation in `veomni/distributed/async_offloading.py`
2. Replaced the #965 `OffloadConfig` fields with `enable_async_activation_offload`
3. Split `BaseTrainer`: `_build_parallelized_model` still reads `enable_async_activation` (the field is gone → `AttributeError`), while `_build_training_context` uses the old global `saved_tensors_hooks`

Even if the YAML is switched to the #965 flag, **Wan’s current GC nesting means #965 will not offload the activations that actually stay resident.**

#965 assumes `async_save_on_cpu` wraps **`GradientCheckpointingLayer.__call__` from the outside**, with checkpoint on the inside. Then `_NoopSaveInputs` saves `hidden_states` while the offload hook is already active.

Wan does the opposite. `WanTransformer3DModel_forward` checkpoints from outside the block:

```python
hidden_states = self._gradient_checkpointing_func(
    _wan_block_ckpt_wrapper(block),
    hidden_states, ...
)
```

The order becomes `checkpoint (outer) → block.__call__ (#965 hook is inner)`. `hidden_states` are saved on the NPU before `__call__` runs. With GC on, those block inputs are the tensors that stay resident; intermediates are recomputed and dropped. Changing only the config will barely reduce I2V-14B memory.

That is why the old path added a global hook plus `resize_(0)`. That design does not match the #965 / FSDP2 contract (class-level `__call__` leakage, process-wide singleton, fully synchronous H2D on NPU). Do not keep it.

## What to change

### 1. Revert the integration to #965

- Delete `veomni/distributed/async_offloading.py`, `docs/key_features/async_activate_offload.md`, and `tests/distributed/test_async_activation_offload.py`
- Restore the `main` `OffloadConfig` fields: `enable_async_activation`, `activation_offload_modules`, `activation_offload_host_cache_limit_gb`, and the mutual-exclusion checks
- Keep `BaseTrainer._build_parallelized_model` on the #965 `apply_async_activation_offload(...)` path (before FSDP)
- Restore `_build_training_context` from `main`: when async offload is on, `fwd/bwd` should be `nullcontext`; do not register delayed-release hooks
- In `torch_parallelize.py`, stop popping `enable_async_activation_offload`, and do not globally patch `_CheckpointFrame.check_recomputed_tensors_match`

### 2. Move Wan GC into the block `__call__`

`WanTransformerBlock` is a **diffusers** class; it is not defined in this repo. Subclass it in VeOmni and put **GCL first** in the MRO, otherwise `__call__` stays `nn.Module.__call__`:

```python
from transformers.modeling_layers import GradientCheckpointingLayer
from diffusers.models.transformers.transformer_wan import (
    WanTransformerBlock as _WanTransformerBlock,
)

class WanTransformerBlock(GradientCheckpointingLayer, _WanTransformerBlock):
    pass
```

diffusers `__init__` still constructs the original class. After `WanTransformer3DModel.__init__`, rebind the instances (GCL adds no instance state):

```python
for block in self.blocks:
    block.__class__ = WanTransformerBlock
```

Keep `_no_split_modules = ["WanTransformerBlock"]` so #965 and FSDP still match.

Then **remove the outer checkpoint** and only call the block (positional args, not `hidden_states=`):

```python
for block in self.blocks:
    hidden_states = block(hidden_states, encoder_hidden_states, timestep_proj, rotary_emb)
```

`_wan_block_ckpt_wrapper` can go. `early_stop` is passed into `torch.utils.checkpoint.checkpoint()` by `gradient_checkpointing_enable`, not into `forward`.

If you only inherit GCL and leave the outer `_gradient_checkpointing_func`, you get double checkpointing and #965 still cannot offload the I2V activations.

### 3. Point the configs at the #965 flag

In `configs/dit/wan2.2_I2V_5B.yaml` and `14B.yaml`:

```yaml
offload_config:
  enable_activation: false
  enable_async_activation: true
  # leave activation_offload_modules empty to use model._no_split_modules
```

`WanTransformer3DModel` already has `_no_split_modules = ["WanTransformerBlock"]`; no extra module list is needed.

### 4. Do not modify the #965 implementation

Leave `async_offload.py` identical to `main`. The NPU path already goes through `veomni.utils.device`. Wan is missing the GC nesting, not a second offload runtime.

---

After this, #965’s hook is outside and GCL’s checkpoint is inside — the same layout as the LLM path — and the old implementation can be deleted entirely.
